metricbeat/module/kubernetes/state_job/state_job.go

```diff
@@ -0,0 +1,156 @@
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package state_job
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/common/kubernetes"
+	p "github.com/elastic/beats/v7/metricbeat/helper/prometheus"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/mb/parse"
+	k8smod "github.com/elastic/beats/v7/metricbeat/module/kubernetes"
+	"github.com/elastic/beats/v7/metricbeat/module/kubernetes/util"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/metrics"
+)
+
+var (
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+	}.Build()
+
+	mapping = &p.MetricsMapping{
+		Metrics: map[string]p.MetricMap{
+			// Make everything in "kube_job_owner" available for use in the Labels section, below.
+			"kube_job_owner": p.InfoMetric(),
+			// These fields are mapped are mapped 1:1 from their KSM metrics.
+			"kube_job_status_active":          p.Metric("pods.active"),
+			"kube_job_status_failed":          p.Metric("pods.failed"),
+			"kube_job_status_succeeded":       p.Metric("pods.succeeded"),
+			"kube_job_spec_completions":       p.Metric("completions.desired"),
+			// completions observed?
+			"kube_job_spec_parallelism":       p.Metric("parallelism.desired"),
+			"kube_job_created":                p.Metric("time.created", p.OpUnixTimestampValue()),
+			"kube_job_status_completion_time": p.Metric("time.completed", p.OpUnixTimestampValue()),
+			// These fields will be set to "true", "false", or "unknown" based on input that looks
+			// like this:
+			//
+			// kube_job_complete{namespace="default",job_name="timer-27074308",condition="true"} 1
+			// kube_job_complete{namespace="default",job_name="timer-27074308",condition="false"} 0
+			// kube_job_complete{namespace="default",job_name="timer-27074308",condition="unknown"} 0
+			"kube_job_complete": p.LabelMetric("status.complete", "condition", p.OpLowercaseValue()),
+			"kube_job_failed":   p.LabelMetric("status.failed", "condition", p.OpLowercaseValue()),
+		},
+
+		Labels: map[string]p.LabelMap{
+			// Jobs are uniquely identified by the combination of name and namespace.
+			"job_name":   p.KeyLabel("name"),
+			"namespace":  p.KeyLabel(mb.ModuleDataKey + ".namespace"),
+			// Add owner information provided by the "kube_job_owner" InfoMetric.
+			"owner_kind":          p.Label("owner.kind"),
+			"owner_name":          p.Label("owner.name"),
+			"owner_is_controller": p.Label("owner.is_controller"),
```

Comment:
This could, in theory, be a proper boolean, but I wasn't sure how to cast it down here in the `Labels` section, where `BooleanMetric` can't be used (as far as I can tell).

Any tips? OK to leave it as a "boolean string", maybe?

Reply:
Since we don't just have true/false but it can also be `<none>` I think we are OK with a "boolean string".

Reply:
Cool, thanks. Would you normally store the exact `"<none>"` string from KSM, or should I translate it to `"none"` or similar?

Reply:
We could "cook" it a little bit and convert it to `"none"` but I think keeping as is work too. I'm more into leaving it as what we get from the original source.

So need to change :).